app/src/main/java/com/gh4a/fragment/RepositoryFragment.java

```diff
@@ -137,7 +102,34 @@ public void onRefresh() {
         if (mImageGetter != null) {
             mImageGetter.clearHtmlCache();
         }
-        hideContentAndRestartLoaders(0, 1);
+
+        this.reloadData();
+    }
+
+    public void reloadData() {
+        setContentShown(false);
+
+        Log.d("TEST", "reloadData callleeeed");
+        loadReadme()
+                .compose(RxTools.handle(getActivity(), RepositoryService.LOAD_README, true))
```

Comment:
RepositoryActivity didn't have these compose method calls. Could you explain why It's only used here?

Reply:
when no **cache refresh** or **loaders restart** is needed, in such cases as for -
- `updateStarring`
- `updateWatching`

services, the `compose` call can be hidden within the **Service** class (`RepositoryService`). 

Still I gotta update the code since : 
- `isStarring`
- `isWatching` 

services **do** need to be refreshed along with **repository** and **readme**. So there will be calls to `compose` within `RepositoryActivity` as well